GraphBase/src/graph/DfsIterator.java

```diff
@@ -0,0 +1,63 @@
+package graph;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Stack;
+
+public class DfsIterator implements Iterator {
```

Comment:
I guess Iterator should have a generic type. In your case, it seems to be Iterator<Vertex>